src/otopi/minidnf.py

```diff
@@ -507,7 +508,10 @@ def __init__(
         if not packager.ok_to_use_dnf():
             raise RuntimeError('minidnf is disabled')
 
-        if int(dnf.__version__.split('.')[0]) not in (2, 3, 4):
+        if LooseVersion(dnf.__version__) < LooseVersion('4.7'):
+            # Recent changes are incompatible with older dnf 4.
+            # See e.g. recent reports on the list about a missing
+            # _get_key_for_package function in dnf 4.2.23.
             raise RuntimeError(_('Incompatible DNF'))
```

Comment:
Maybe add to the error message which versions are supported

Reply:
Done